### ISSUE TITLE
Adding custom plots to Dask Dashboard 

### ISSUE THREAD

**Issue by mronda** [dask/distributed#3503]
Hi, 
I am running multiple tasks that takes a while to return, so I want to be able to graph some custom data in between to monitor their progress. Can you guys point me to the documentation/code so I can see where to make changes to the Bokeh Dashboard? Thanks in advance! 
-Max

**Comment by quasiben** [comment]
@mronda I don't think we have docs on customizing the dashboard (though that would be welcome).  Instead, I would recommend reading through 2 recent pull requests:

- https://github.com/dask/distributed/pull/3243/
- https://github.com/dask/distributed/pull/3361

Can I ask what additional data you were hoping visualize ?

**Comment by mronda** [comment]
Hi @quasiben , thanks for the response. I checked the first pr but still not getting how I should connect to that server and pass, from my script, the data to be received by Bokeh server. What handler would I need in my script to make that communication? Not too familiar with Bokeh, any help would be a huge time saver right now. 
Oh and I am trying to plot the moving average of a model for monitoring purposes. 
Thanks in advance ! 
-Max

**Comment by jacobtomlinson** [comment]
I wonder if a section of the dashboard showing [published datasets](https://distributed.dask.org/en/latest/publish.html) would be useful? If a dataset is already persisted and has a type of `int`, `float` or `str` it could previewed in the dashboard too. 

That way you could continuously persist the moving average and view it in the dashboard?

Perhaps this is abusing the published datasets feature a little too much though...

**Comment by mronda** [comment]
Hi @jacobtomlinson , So I've tried using `client.publish_datasets` to do some visualization of my data. The problem I encountered, and correct me if I am wrong, is that I needed to constantly unpublish and publish the data as I could not persist it to the client. I've also tried using a dask.distributed Queue, which worked fine but I am looking into a more pipeline approach where I can use the dashboard provided to monitor the running average. Any recommendations? Maybe a better way to use those collections? Thanks!
-Max

**Comment by mrocklin** [comment]
@lesteve had something for this.  I think he submitted a pull request in
the last year or so.  I can't find it in the documentation right now, but
you might search through closed github pull requests for things authored by
him and my guess is that you'll come across it.

**Comment by quasiben** [comment]
I think this is the PR, https://github.com/dask/distributed/pull/2169, @mrocklin is referring to.
